python-312/PKGBUILD

```diff
@@ -0,0 +1,91 @@
+# Maintainer: Tobias Kunze <r@rixx.de>
+# Maintained at https://github.com/rixx/pkgbuilds, feel free to submit patches
+# Updated for CachyOS: Laio Seman <laio [at] ieee.org>
+
+pkgname=python312
+pkgver=3.12.1
+pkgrel=3
+_pybasever=3.12
+_pymajver=3
+pkgdesc="Major release 3.12 of the Python high-level programming language"
+arch=('x86_64')
+license=('custom')
+url="https://www.python.org/"
+depends=('bzip2' 'expat' 'gdbm' 'libffi' 'libnsl' 'libxcrypt' 'openssl' 'zlib')
+makedepends=('bluez-libs' 'mpdecimal' 'gdb' 'tk', 'llvm-git')
+optdepends=('sqlite' 'mpdecimal: for decimal' 'xz: for lzma' 'tk: for tkinter')
+source=(https://www.python.org/ftp/python/${pkgver}/Python-${pkgver}.tar.xz)
+sha256sums=('8dfb8f426fcd226657f9e2bd5f1e96e53264965176fa17d32658e873591aeb21')
+validpgpkeys=(
+    '0D96DF4D4110E5C43FBFB17F2D347EA6AA65421D' # Ned Deily (Python release signing key) <nad@python.org>
+    'E3FF2839C048B25C084DEBE9B26995E310250568' # Łukasz Langa (GPG langa.pl) <lukasz@langa.pl>
+)
+
+prepare() {
+    cd "${srcdir}/Python-${pkgver}"
+
+    # FS#23997
+    sed -i -e "s|^#.* /usr/local/bin/python|#!/usr/bin/python|" Lib/cgi.py
+
+    # Ensure that we are using the system copy of various libraries (expat, zlib and libffi),
+    # rather than copies shipped in the tarball
+    rm -rf Modules/expat
+    rm -rf Modules/_decimal/libmpdec
+}
+
+build() {
+    cd "${srcdir}/Python-${pkgver}"
+
+    endianness=$(python -c "import sys; print('little' if sys.byteorder == 'little' else 'big')")
+
+    # Set the ax_cv_c_float_words_bigendian variable based on the Python output
+    if [ "$endianness" == "little" ]; then
+        export ax_cv_c_float_words_bigendian=no
+    else
+        export ax_cv_c_float_words_bigendian=yes
+    fi
```

Comment:
Well, if this detection is at buildtime (in our case on zen4 machine) people with big little wont benefit from it

Reply:
Python 3.12 refuses to compile without it being explicity.